src/app/errorReports/newErrorReportForm/NewErrorReportForm.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { postErrorReport, getLocation } from '../../../actions';
+import { selectIsPostingErrorReport } from '../../../reducers';
+import { selectLocationError, selectLocationData } from '../../../selectors/location';
+import LoadingLabel from '../../../components/form/LoadingLabel';
+import ErrorLabel from '../../../components/form/ErrorLabel';
+import ErrorReportText from './ErrorReportText';
+import ErrorReportServices from './ErrorReportServices';
+
+class NewErrorReportForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      errorReportText: '',
+      errorReportGeneralLocationError: false,
+      errorReportServices: [],
+      isErrorReportFinished: false,
+    };
+
+    this.onErrorReportTextChanged = this.onErrorReportTextChanged.bind(this);
+    this.onErrorReportServicesChanged = this.onErrorReportServicesChanged.bind(this);
+    this.onErrorReportGeneralLocationErrorChanged = this.onErrorReportGeneralLocationErrorChanged.bind(this);
+    this.onErrorReportFinished = this.onErrorReportFinished.bind(this);
+    this.onErrorReportSubmitted = this.onErrorReportSubmitted.bind(this);
+  }
+
+  componentDidMount() {
+    if (!this.props.locationData) {
+      this.props.getLocation(this.props.match.params.locationId);
```

Comment:
To simplify the code of the component itself, I'd move the knowledge of how to get locationId out of here and into where getLocation is defined below.
This would also make it easier if you ultimately want to split the more "container-y" parts of this file into an actual separate container component.

See e.g. LocationDetailsContainer.js vs LocationDetails.jsx.